packages/query/src/stacks/balance/account-balance.hooks.ts

```diff
@@ -55,13 +53,45 @@ export function useStxCryptoAssetBalance(address: string) {
       client,
       network: network.chain.stacks.url,
     }),
-    select: resp =>
-      createStxCryptoAssetBalance(createStxMoney(resp), inboundBalance, outboundBalance),
+    select: resp => createStxMoney(resp),
+  });
+}
+
+export function useStxCryptoAssetBalance(address: string) {
+  const client = useStacksClient();
+  const network = useCurrentNetworkState();
+
+  const initialBalanceQuery = useStxBalanceQuery(address);
+
+  const defaultPendingBalance = createMoney(0, 'STX');
+  const { balance: inboundBalance = defaultPendingBalance, query } = useMempoolTxsInboundBalance(address);
+  const { balance: outboundBalance = defaultPendingBalance } = useMempoolTxsOutboundBalance(address);
+
+  const processedBalanceQuery = useQuery({
+    ...createGetStacksAccountBalanceQueryOptions({
+      address,
+      client,
+      network: network.chain.stacks.url,
+    }),
+    select: resp => {
+      const initialBalance = createStxMoney(resp);
+      return createStxCryptoAssetBalance(initialBalance, inboundBalance, outboundBalance);
+    },
+    enabled: !!initialBalanceQuery.data,
   });
+
+  return {
+    initialBalanceQuery,
+    processedBalanceQuery,
+    isLoadingAdditionalData: query.isLoading,
+  };
 }
 
 export function useStxAvailableUnlockedBalance(address: string) {
-  return useStxCryptoAssetBalance(address).data?.unlockedBalance ?? createMoney(0, 'STX');
+  return (
+    useStxCryptoAssetBalance(address).processedBalanceQuery.data?.unlockedBalance ??
+    createMoney(0, 'STX')
+  );
```

Comment:
```suggestion
  const stxBalance = useStxCryptoAssetBalance(address);
  return (
    stxBalance.processedBalanceQuery.data?.unlockedBalance ??
    createMoney(0, 'STX')
  );
```

also `processedBalanceQuery` is odd name to export from a hook I think?

Reply:
What is the relevance that it's been filtered or processed?